docs/question.md

```diff
@@ -447,3 +447,90 @@ For most [elements] there are four different ways of grading the student answer.
+
+```python
+import random
+def generate(data):
+  # It is recommended to use .get() with a default value in case params have not been set, for example,
+  # when viewing the question from the Questions list.
+  param_names = data['params'].get('names', [{'name': 'Matt', 'photo': '001'}, {'name': 'Craig', 'photo': '002'}])
+  # Choose a random person to display.
+  chosen = random.choice(param_names)
+  data['params']['photo'] = chosen['photo']
+  data['correct_answers']['name'] = chosen['name']
+```
+
+Any `questionParams` that applies to this question and has a `names` key will provide the list to use as the parameter `names`; otherwise, the default list `[{'name': 'Matt', 'photo': '001'}, {'name': 'Craig', 'photo': '002'}]` will be used.
+
+The chosen name could be used to generate some HTML in `question.html`:
+
+```html
+What is this person's first name?
+
+<pl-figure
+  file-name="student_images/{{params.photo}}.png"
+  width="140px"
+  directory="clientFilesCourse"
+></pl-figure>
+
+<pl-string-input answers-name="name"></pl-string-input>
+```
+
+When a new semester begins and a course instance is created, the `courseInstance.json` can be updated with the names of the course instructor and TAs.
+
+```json
+{
+  "uuid": ...,
+  ...,
+  "questionParams": {
+    "names": [{"name": "Neal", "photo": "003"}, {"name": "Jonatan", "photo": "004"}]
+  }
+}
+```
+
+When the question is used in an assessment for this course instance, Neal and Jonatan will be shown in the HTML instead of Matt and Craig.
+
+The question can be used in a second assessment with a different set of names by setting `questionParams` in the `infoAssessment.json` file:
+
+```json
+{
+  "uuid": ...,
+  ...,
+  "zones": [
+    {
+      "questions": [
+        {
+          "id": "myPhotoMatchingQuestion",
+          "points": 5,
+          "questionParams": {
+            "names": [{"name": "Luke", "photo": "005"}, {"name": "Leia", "photo": "006"}]
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+Doing this does not change the question in the original assessment, which would not be the case if the `server.py` was updated with the new list of names.
+
+Question params can reduce the need to duplicate questions in order to change the data source of its variation. This makes maintenance of the questions easier. Examples:
+
+- A "match each nation with its capital city" question can use a different set of nations depending on what region has been studied most recently. (Instead of different questions for each region.)
+- A question on programming language syntax errors can involve more examples of errors as new features with different syntax rules are learned. (Instead of different questions for later in the course.)
+- A question that generates an indefinite integrals can use simple trig functions on an early calculus exam, and later use inverse trig functions. (Instead of two different questions that only vary in the chosen function.)
```

Comment:
```suggestion
- A question that generates an indefinite integral can use simple trig functions on an early calculus exam, and later use inverse trig functions. (Instead of two different questions that only vary in the chosen function.)
```